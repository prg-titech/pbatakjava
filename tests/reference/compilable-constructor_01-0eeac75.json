{
    "basename": "compilable-constructor_01-0eeac75",
    "cmd": "{infile}",
    "infile": "tests/constructor_01/build.sh",
    "infile_hash": "035418d5e91eec172eb7ff1246c1f004621ab5d54c99b281752146a2",
    "stdout": "compilable-constructor_01-0eeac75.stdout",
    "stdout_hash": "a679a72787e27d34a1af01379e1d630bd8f965688fe534dd2dce6aa6",
    "stderr": null,
    "stderr_hash": null,
    "returncode": 0
}