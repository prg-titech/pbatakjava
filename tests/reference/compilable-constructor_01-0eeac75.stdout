
public interface Point {
}

public interface Point_Factory {
  public Point make(int x, int y);

  public Point make(int x, int y, int z);
}

public class Point_v1 implements Point {
  public int x;

  public int y;

  public Point_v1(int x, int y) {
    this.x = x;
    this.y = y;
  }
}

public class Point_v1_Factory implements Point_Factory {
  public Point make(int x, int y) {
    return new Point_v1(x, y);
  }

  public Point make(int x, int y, int z) {
    return null;
  }
}

public class Point_v2 implements Point {
  public int x;

  public int y;

  public int z;

  public Point_v2(int x, int y, int z) {
    this.x = x;
    this.y = y;
    this.z = z;
  }
}

public class Point_v2_Factory implements Point_Factory {
  public Point make(int x, int y, int z) {
    return new Point_v2(x, y, z);
  }

  public Point make(int x, int y) {
    return null;
  }
}
